tsl/src/remote/connection.c

```diff
@@ -441,7 +442,7 @@ handle_result_create(PGEventResultCreate *event)
 	TSConnection *conn = PQinstanceData(event->conn, eventproc);
 	ResultEntry *entry;
 
-	Assert(NULL != conn);
+	Ensure(conn, "no instance data available");
```

Comment:
Same  here, this should be an `Assert()`. The connection is allocated elsewhere and that's where we should have the check.